bots/archive/bot.py

```diff
@@ -0,0 +1,162 @@
+import csv
+import datetime
+import gzip
+import io
+import logging
+import os
+
+import boto3
+import botocore
+
+from django.db import transaction
+from django.conf import settings
+
+from share.bot import Bot
+from share.models import CeleryProviderTask
+
+logger = logging.getLogger(__name__)
+
+
+class ArchiveBot(Bot):
+
+    def run(self):
+        # check for storage settings
+        if not settings.AWS_ACCESS_KEY_ID or not settings.AWS_SECRET_ACCESS_KEY:
+            if os.environ.get('DEBUG', True) is False:
+                raise Exception('No storage found! CeleryTasks will NOT be archived or deleted.')
+            logger.warning('No storage found! CeleryTasks will NOT be archived but WILL be deleted.')
+
+        logger.info('%s started converting queryset to csv data at %s', self.started_by, datetime.datetime.utcnow().isoformat())
+
+        if not settings.CELERY_TASK_BUCKET_NAME:
+            raise Exception('Bucket name not set! Please define bucket name in project.settings')
+        bucket = self.get_bucket(settings.CELERY_TASK_BUCKET_NAME)
+
+        current_time = datetime.datetime.utcnow()
+        one_day = datetime.timedelta(days=-1)
+        one_week = datetime.timedelta(weeks=-1)
+        two_weeks = datetime.timedelta(weeks=-2)
+
+        # bots.elasticsearch (24hrs)
+        self.elasticsearch_tasks = CeleryProviderTask.objects.filter(
+            app_label='elasticsearch',
+            timestamp__lt=current_time + one_day
+        )
+        # normalizertask (1 week)
+        self.normalizer_tasks = CeleryProviderTask.objects.filter(
+            name='share.tasks.NormalizerTask',
+            timestamp__lt=current_time + one_week
+        )
+        # disambiguatortask (1 week)
+        self.disambiguator_tasks = CeleryProviderTask.objects.filter(
+            name='share.tasks.DisambiguatorTask',
+            timestamp__lt=current_time + one_week
+        )
+        # harvestertask (2 weeks)
+        self.harvester_tasks = CeleryProviderTask.objects.filter(
+            name='share.tasks.HarvesterTask',
+            timestamp__lt=current_time + two_weeks
+        )
+        # archivetask (2 weeks)
+        self.archive_tasks = CeleryProviderTask.objects.filter(
+            name='share.tasks.ArchiveTask',
+            timestamp__lt=current_time + two_weeks
+        )
+
+        if self.elasticsearch_tasks.exists():
+            compressed_data = self.queryset_to_compressed_csv(self.elasticsearch_tasks)
+            self.put_s3(bucket, 'elasticsearch/elasticsearch_tasks_', compressed_data)
+
+            logger.info('Finished archiving data for Elasticsearch CeleryTask at %s', datetime.datetime.utcnow().isoformat())
```

Comment:
Ohh, awk. I was just formatting the logs like the ones in `DisambiguatorTask`, `ApplySingleChangeSet`, and `ApplyChangeSets` in [tasks.py](https://github.com/CenterForOpenScience/SHARE/blob/develop/share/tasks.py#L194).